packages/zig/preinst.sh.in

```diff
@@ -0,0 +1,44 @@
+#!@TERMUX_PREFIX@/bin/sh
+# shellcheck shell=sh
+
+echo "Checking that your device meets the"
+echo "minimum kernel version for Zig @TERMUX_PKG_VERSION@"
+echo
+
+is_supported() {
+	_KERNEL_VERSION="$1"
+	_MINIMUM_VERSION="$2"
+	# shellcheck disable=SC2194 # We replace this in the build script
+	case "@TERMUX_PACKAGE_FORMAT@" in
+		apt)
+			if dpkg --compare-versions "$_KERNEL_VERSION" lt "$_MINIMUM_VERSION"; then
+				return 1
+			fi
+		;;
+		pacman)
+			if [ "$(vercmp "$_KERNEL_VERSION" "$_MINIMUM_VERSION")" != 1 ]; then
```

Comment:
This will return 0 if kernel version is 3.16.0, should probably check if it is -1 instead